src/Raven.Server/RavenServerStartup.cs

```diff
@@ -157,23 +159,90 @@ private async Task RequestHandler(HttpContext context)
                 context.Response.StatusCode = (int)HttpStatusCode.OK;
                 context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
                 context.Response.Headers[Constants.Headers.ServerVersion] = RavenVersionAttribute.Instance.AssemblyVersion;
+                // Check context url AND TW clients
+                bool isQueryBool = (TrafficWatchManager.HasRegisteredClients && IsQuery(context));
+
+                if (isQueryBool)
+                {
+                    // init new MS to save context.request.body
+                    MemoryStream memoryStreamForRequest = new MemoryStream();
+                    // copy stream
+                    await context.Request.Body.CopyToAsync(memoryStreamForRequest);
+                    // put memory stream pointer to start position
+                    memoryStreamForRequest.Position = 0;
+                    // add memory stream to context
+                    context.Items.Add("requestMemoryStream", memoryStreamForRequest);
+                    // copy memory stream back to context.request.body
+                    context.Request.Body = memoryStreamForRequest;
+                }
 
                 var sp = Stopwatch.StartNew();
+                
                 database = await _router.HandlePath(context, context.Request.Method, context.Request.Path.Value);
-                sp.Stop();
 
                 if (_logger.IsInfoEnabled && SkipHttpLogging == false)
                 {
                     _logger.Info($"{context.Request.Method} {context.Request.Path.Value}?{context.Request.QueryString.Value} - {context.Response.StatusCode} - {sp.ElapsedMilliseconds:#,#;;0} ms");
                 }
-
-                if (TrafficWatchManager.HasRegisteredClients)
-                    LogTrafficWatch(context, sp.ElapsedMilliseconds, database);
+                // if query OR multi-get OR index AND TW has clients
+                if (isQueryBool)
+                {
+                    string customInfoString = "N/A";
+                    if (context.Request.Method == "GET")                          // GET method and URL has query <=> query from Raven.Studio
+                    {
+                        // get the query value
+                        StringValues value;
+                        customInfoString = context.Request.Query.TryGetValue("query", out value) ? value.ToString() : "N/A";
+                    }
+                    else if (context.Request.Method == "POST")
+                    {
+                        try
+                        {
+                            // object for the memory stream stored at the context
+                            object val;
+                            // if there is a ContentType
+                            if (context.Request.ContentType == "application/json; charset=UTF-8")
```

Comment:
why we have those two cases?

I think you should use: RequestBodyStream() method in both cases?